src/program/lwaftr/migrate_configuration/migrate_configuration.lua

```diff
@@ -270,13 +270,18 @@ function load_binding_table(file)
    return parse_binding_table(Parser.new(source:as_text_stream()))
 end
 
+
 local function config_to_string(schema, conf)
    if type(schema) == "string" then
       schema = yang.load_schema_by_name(schema)
    end
-   local memfile = util.string_output_file()
-   yang.print_config_for_schema(schema, conf, memfile)
-   return memfile:flush()
+   -- To keep memory usage as low as possible write it out to a temp file.
+   local temp = io.open("/tmp/migrate-configuration-temp", "w")
+   yang.print_config_for_schema(schema, conf, temp)
+   temp:close()
+   conf = io.open("/tmp/migrate-configuration-temp", "r"):read("*a")
+   os.remove("/tmp/migrate-configuration-temp")
+   return conf
```

Comment:
So I'm still a bit sketchy on how luajit's GC actually works but, it seemed after each migration the memory usage almost doubled. Certain migrations should cause a higher memory usage due to the nature of the config changes but not to the degree I was seeing them. It seemed that this function was largely responsible.

If I look at how it works with `memfile` if I explicitly collect the garbage where it is now so it should do a full GC collect after each migration and then print to tell me how much memory it's using before each migration it looks like this:
```
$ sudo ./snabb lwaftr migrate-configuration -f legacy lwaftr.conf
Migrating version:	legacy	(memory: 9.0424470901489)
Migrating version:	3.0.1	(memory: 462.83460521698)
Migrating version:	3.0.1.1	(memory: 430.83438396454)
Migrating version:	3.2.0	(memory: 730.37660884857)
```

config_to_string function:
```lua
local function config_to_string(schema, conf)
   if type(schema) == "string" then
      schema = yang.load_schema_by_name(schema)
   end
   local memfile = util.string_output_file()
   yang.print_config_for_schema(schema, conf, memfile)
   local migrated = memfile:flush()
   memfile = nil
   conf = nil
   return migrated
end
```

I would have expected without any of the `memfile = nil` and `config = nil` that when the function returns and the GC is explicitly called to collect then it'd free those. I go the step further and set them to nil and it still seems to balloon the memory.

What it looked like when I used the temp file writing that's currently in this PR:
```
$ sudo ./snabb lwaftr migrate-configuration -f legacy lwaftr.conf
Migrating version:	legacy	(memory: 9.0434036254883)
Migrating version:	3.0.1	(memory: 166.33897018433)
Migrating version:	3.0.1.1	(memory: 134.27617263794)

```

I don't see why in the first example after `config_to_string` has returned and presumably `memfile` can be freed it still uses 462 MB rather than the 166 MB that the second version has. I can find any tooling to do memory profiling, I'm a bit lost to why this is happening.

It seems there are two solutions:

1) Figure out why the GC isn't behaving as I think it ought to and thus how to fix it (ideal but will probably take quite a while)
2) Put something unique like exact timestamp + random token in the file name so multiple migrations can be run at once.

Any ideas?